### PR TITLE
Implement SuperFind

### LINKED ISSUE #129: Implement SuperFind class
Allow user to find multiple types of field in the same command

### PR DESCRIPTION
**Pull request by yooplo**
This would allow the user to have multiple fields in one Find command
(e.g. find n/John t/friend p/95631234)

Changes include:
- Refactoring of AbstractFindCommand to SuperFind
- Implementing the relevant test cases
- Add SuperFind to UML diagram

Closes #129

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -99,6 +99,6 @@ public void setGuiSettings(GuiSettings guiSettings) {
     }
 
     private boolean shouldSaveCampusConnect(Command c) {
-        return !(c.equals(new UndoCommand()) || c instanceof ListCommand || c instanceof AbstractFindCommand);
+        return !(c instanceof UndoCommand || c instanceof ListCommand || c instanceof SuperFindCommand);
```
Comment (CYX22222003) [inline src/main/java/seedu/address/logic/LogicManager.java:102]: This potentially conflicts with the PR #140 made by @chrisjohntan

**Review by CYX22222003 — COMMENTED** [review]
LGTM. The feature seems to work as what we expect.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 2]
```diff
@@ -7,57 +7,63 @@
...
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_TAG);
+        ArgumentMultimap argMultimap = ArgumentTokenizer
+                .tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_TAG);
+
+        List<Predicate<Person>> predicates = new ArrayList<>();
 
         if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
-            tempList = argMultimap.getAllValues(PREFIX_NAME);
-            return new FindByNameCommand(new NameContainsKeywordsPredicate(tempList));
+            List<String> nameKeywords = argMultimap.getAllValues(PREFIX_NAME);
```
Comment (chrisjohntan) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:42]: Perhaps we could see if we can extract these prefix checks into their own methods to separate the level of abstraction.

**Review by chrisjohntan — APPROVED** [review]
LGTM.

**Pull request merged**